package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -1,79 +1,124 @@
-"""`kedro_viz.data_access.repositories.modular_pipelines`
-defines repository to centralise access to modular pipelines data."""
+"""`kedro_viz.data_access.repositories.modular_pipelines` defines
+repository to centralise access for modular pipelines data."""
 
-from typing import Dict, Optional, Union
+import hashlib
+from typing import Dict, List, Set, Tuple, Union
+
+from kedro.pipeline import Pipeline as KedroPipeline
+from kedro.pipeline.node import Node
 
 from kedro_viz.constants import ROOT_MODULAR_PIPELINE_ID
 from kedro_viz.models.flowchart import (
-    DataNode,
     GraphNode,
     GraphNodeType,
     ModularPipelineChild,
     ModularPipelineNode,
-    ParametersNode,
-    TranscodedDataNode,
 )
+from kedro_viz.utils import TRANSCODING_SEPARATOR, _strip_transcoding, is_dataset_param
+
+
+def _hash(value: str):
+    return hashlib.sha1(value.encode("UTF-8")).hexdigest()[:8]
+
+
+def _hash_input_output(item: str) -> str:
+    """Hash the input/output dataset."""
+    return (
+        _hash(_strip_transcoding(item))
+        if TRANSCODING_SEPARATOR in item
+        else _hash(item)
+    )

```

Comment:
I would put these in `utils` and import them here, rather than the transcoding logic. Hide as much as possible behind a smaller API footprint.